tuf/data/targets.go

```diff
@@ -23,6 +25,38 @@ type Targets struct {
 	Delegations Delegations `json:"delegations,omitempty"`
 }
 
+// isValidTargets returns an error, or nil, depending on whether the content of the struct
+// is valid for targets metadata.  This does not check signatures or expiry, just that
+// the metadata content is valid.
+func isValidTargets(t Targets, roleName string) error {
+	// even if it's a delegated role, the metadata type is "Targets"
+	expectedType := TUFTypes[CanonicalTargetsRole]
+	if t.Type != expectedType {
+		return ErrInvalidMeta{
+			Role: roleName, Msg: fmt.Sprintf("expected type %s, not %s", expectedType, t.Type)}
+	}
+
```

Comment:
Should we validate `roleName` as well, before iterating through the delegation roles?